app/models/friend_request.rb

```diff
@@ -0,0 +1,31 @@
+class FriendRequest < ApplicationRecord
+  belongs_to :sender, class_name: 'User'
+  belongs_to :receiver, class_name: 'User'
+ 
+  # validate :not_friends
+  validate :not_pending
+  validate :not_self
+  validate :not_friends
+
+  def accept
+    sender.basic_friends << receiver
```

Comment:
We still need some more improvement here. As I created a friend request and ran this `accept` method on it I observe that the friendship that was created only reflected in the `sender` 's side but not on the `receiver` 's side. 

Here's a suggestion. You probably dont' need to have this method of accepting a friend request. What you can probably have is a callback in your friendship model that deletes any associated request whenever a friendship record is created. 

For instance I have two users `ryto` and `mike`. `ryto` sent a request to `mike`. Then when `mike` accepts the friendship then definitely this method will run in the `Frienship` model:

`Friendship.create(user_id: ryto.id, friend_id: mike.id)`

What you want is that upon the above method in `Friendship` finishing (`after_create` or `after_save`) the request between `ryto` and `mike` to be deleted.